lib/faraday/log_filter/filter.rb

```diff
@@ -0,0 +1,40 @@
+module Faraday
+  module LogFilter
+    class Filter
+      def initialize(params_to_filter)
+        @params_to_filter   = Array(params_to_filter)
+        @params_to_truncate = @params_to_filter.last.is_a?(Hash) ? @params_to_filter.pop : {}
+      end
+
+      def filter_url(url)
+        filtered_query = filter(CGI::parse(url.query.to_s))
+        url.query      = filtered_query.map { |params| params.join("=") } * "&"
```

Comment:
Looks like the [ruby style guide](https://github.com/bbatsov/ruby-style-guide#array-join) recommends using `join` instead of `*`